pinconfigs/gd32l233xx.yml

```diff
@@ -0,0 +1,977 @@
+
+# GD32L233XX pin definitions
+#
+# Sources:
+# - GD32L233XX Datasheet (Revision 1.2)
+#
+# Pin codes:
+#
+# - 32 pins: Q (GD32L233Kx-QFN32)
+# - 32 pins: K (GD32L233Kx-LQFP32)
+# - 48 pins: C
+# - 64 pins: R
+#
+# Memory codes:
+#
+# - 64Kb   Flash, 16Kb  SRAM: 8
+# - 128Kb  Flash, 24Kb  SRAM: B
+# - 256Kb  Flash, 32Kb  SRAM: C
+#
+# Copyright (c) 2021 BrainCo Inc.
+# SPDX-License-Identifier: Apache 2.0
+
+model: af
+
+series: gd32l233
+
+variants:
+  - pincode: Q
+    memories: [8, B]
+  - pincode: K
+    memories: [8, B]
+  - pincode: C
+    memories: [8]
+  - pincode: C
+    memories: [B, C]
+  - pincode: R
+    memories: [8]
+  - pincode: R
+    memories: [B, C]
+
+signal-configs:
+  COM0:
```

Comment:
Although it is a matter of preference in writing style, It may be easier to maintain if you don't create multiple tables. In other words, pin exclusions should be written beside the pin definition.
This make to keep one `pins` table.

For example, as below.

```
pins:
  PA8:
    afs:
      COM0: 3
      CTC_SYNC: 8
    excludes-pincodes:
      COM0: [Q, K, C]
```

I think there is no problem because it works with the current code. However, please consider if it suits your preference.

Reply:
The `exclude-pincodes` and `exclude-memories` in `signal-configs`  is used to hold the perpherals difference. Like `UART4`, it's only available for `GD32L233RB/C` and `GD32L233CB/C`. 

If we move signal configs to pins, this information may not be easily to take out.